src/librustdoc/clean/mod.rs

```diff
@@ -1696,6 +1697,23 @@ impl Clean<Visibility> for hir::Visibility<'_> {
     }
 }
 
+impl Clean<Visibility> for ast::Visibility {
+    fn clean(&self, cx: &mut DocContext<'_>) -> Visibility {
+        match self.kind {
+            ast::VisibilityKind::Public => Visibility::Public,
+            ast::VisibilityKind::Inherited => Visibility::Inherited,
+            ast::VisibilityKind::Crate(_) => {
+                let krate = DefId::local(CRATE_DEF_INDEX);
+                Visibility::Restricted(krate)
+            }
+            ast::VisibilityKind::Restricted { id, .. } => {
+                let did = cx.enter_resolver(|r| r.local_def_id(id)).to_def_id();
```

Comment:
It's unfortunate that this has to call into the resolver :/ I'm hoping to get rid of it at some point: https://github.com/rust-lang/rust/issues/83761. But we already call `load_macro_untracked` above, so hopefully fixing that will allow removing this impl at the same time.